src/Nox.Lib/Extensions/HttpRequestExtensions.cs

```diff
@@ -1,21 +1,29 @@
 ﻿using Microsoft.AspNetCore.Http;
-
+using Nox.Exceptions;
 using System.Net.Http.Headers;
 
 namespace Nox.Extensions;
 
 public static class HttpRequestExtensions
 {
-    public static System.Guid? GetDecodedEtagHeader(this HttpRequest request)
+    public static Guid GetDecodedEtagHeader(this HttpRequest request)
     {
         var ifMatchValue = request.Headers.IfMatch.FirstOrDefault();
-        string? rawEtag = ifMatchValue;
+
+        if (string.IsNullOrEmpty(ifMatchValue))
+        {
+            throw new ConcurrencyException("ETag is empty. ETag should be provided via the If-Match HTTP Header.");
```

Comment:
we need to distinguish this exceptions, because if the etag is not well formed, the status code should be 400

if the Etag condition fails should be 412, needs to be clear to the user what is the type of exception

Should we add a status code to our exceptions ,if its not 500 that, we want to raise ? and set on an interface to be handled by the middleware INoxHandledException